### PR TITLE
Milestone 3 - Game logic

### PR DESCRIPTION
**Pull request by guirecordon**
We split our TicTacToe class which has always been in the main file into three different classes in total. We added the classes Player and Board each into their separate files called "player.rb" and "board.rb" respectively. 
All the classes were instantiated inside of each other every time their methods need to be called and the game is working exactly as it was before.

### CODE REVIEW

File: bin/main [hunk 1]
```diff
@@ -1,112 +1,34 @@
...
+      break if @player.win? or @player.tie?
 
+      round += 1
+     end
+   end
 end
 
+require_relative '../lib/player.rb'
+require_relative '../lib/board.rb'
+require_relative '../lib/game_logic.rb'
```
Comment (Dandush03) [inline bin/main]: please put this at the top of your document

File: lib/player.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    @winning_streaks =
+      [[1, 2, 3],
+       [4, 5, 6],
+       [7, 8, 9],
+       [1, 4, 7],
+       [2, 5, 8],
+       [3, 6, 9],
+       [1, 5, 9],
+       [3, 5, 7]]
+  end
```
Comment (Dandush03) [inline lib/player.rb:18]: I'm confident you can make this long code into a simple and more elegant way with some loops

File: lib/player.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
+class Player
+  attr_accessor :arr_player1, :arr_player2, :player1, :player2
```
Comment (Dandush03) [inline lib/player.rb]: Use attr_accessor only when needed as explained above

File: lib/board.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+class Board
+  attr_accessor :player, :numbers, :arr_player1, :arr_player2
```
Comment (Dandush03) [inline lib/board.rb]: Use attr_accessor only when needed as explained above

File: lib/player.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    end
+  end
+
+  def win?
+    @winning_streaks.each do |streaks|
+      if (@arr_player1 & streaks).to_a.sort == streaks
+        puts 'Player 1 wins!'
+        return true
+      elsif (@arr_player2 & streaks).to_a.sort == streaks
+        puts 'Player 2 wins!'
```
Comment (Dandush03) [inline lib/player.rb]: no `puts` allowed in this file

File: lib/player.rb [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        return true
+      end
+    end
+    false
+  end
+
+  def tie?
+    if @arr_player1.size + @arr_player2.size == 9 and win? == false
+      puts "It's a tie! Play again."
```
Comment (Dandush03) [inline lib/player.rb]: no `puts` allowed in this file

File: lib/board.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  def initialize
+    @numbers = (1..9).to_a
+    @player = Player.new
+  end
+
+  def display
+    print "\n#{@numbers[0]} | #{@numbers[1]} | #{@numbers[2]}
```
Comment (Dandush03) [inline lib/board.rb]: no `print` allowed in this file

**Review by Dandush03 — COMMENTED** [review]
# Statush: :red_circle: Required Changes :red_circle: 
## Great Attemp on this one

### There are some changes you must do before submitting for another review
- [ ] PLEASE RENAME YOUR MAIN FILE TO MAIN.RB SINCE LINTER IS NOT CHECKING FOR LINTERS ERROR ON IT
 > Stickler got updated and with it, they change how files are reviews. so Rubocop on stickler will only review your file if it has it an extension. I hope you understand it's not an issue from our end but it's a most.
- [ ] Use attr_accessors only when there is no other option to it. Don't forget you can always create a function to set or get a value.
- [ ] No use of Puts/Print outside main.rb

## here below I will add some more comments

## Happy Coding :computer: & GL

##### Daniel Laloush

**Review by MarioDena — CHANGES_REQUESTED** [review]
Hello! Good job implementing the requested changes :medal_military: , however your commit messages are lacking.

## Changes required 

- [x] Please change your commit message on commit `451ba72` right now it reads "attr_reader" This message is not descriptive or meaningful, please remember this is required on all projects. We need to be careful of following this basic rules and do not forget about them. This is because when you came back to a big project you need to have descriptive messages in case you want to rollback.

## Optional Changes

- [ ] I would update the read.me file a little bit on every milestone but of course this is a recommendation and not mandatory [OPTIONAL]

After making the requested changes please re-submit your project for re-review.

Cheers! :+1:

**Comment by Anna-Myzukina** [comment]
## Hi team ! Very nice project! :wave: 

# You did a great job, approved  :+1:

* NOTE Ruby methods automatically returns the last evaluated statement therefore, there is no need for explicit declaration `return`

### Go and merge to development.
`Goodluck in your last milestone Happy coding ` <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Pull request merged**